### ISSUE TITLE
terra-form-select Incorrect focus location after pressing TAB / Shitf+TAB on the opened dropdown

### ISSUE THREAD

**Issue by bkrsz** [cerner/terra-core#3570]
# Bug Report

## Description

 - When the dropdown/select box is opened, navigating from the element with the TAB button is not working as expected. 

## Steps to Reproduce

For TAB key:
1. Open https://engineering.cerner.com/terra-ui/components/terra-form-select/form-select/select
2. Navigate to the First example select box on the page and open it
3. Press TAB
4. Press TAB again

Result: 
On step 3., the dropdown is closed but the focus is "lost". 
On step 4., on the second Tab key press, the navigation starts over on the top of the page.

For Shift + TAB key:
1. Open https://engineering.cerner.com/terra-ui/components/terra-form-select/form-select/select
2. Navigate to the first example select box on the page and open it
3. Press Shift + TAB
4. Press Shift + TAB again

Result: 
On step 3., the dropdown is closed but the focus is "lost". 
On step 4., on the second Shift + Tab key press, the navigation starts at the bottom of the page.

## Additional Context / Screenshots

Initial state (opened dropdown):
![image](https://user-images.githubusercontent.com/53014693/148972802-c820c6ec-bcf3-427a-9471-eaa0bc9f77c5.png)

State after pressing TAB twice when the dropdown is opened:
![image](https://user-images.githubusercontent.com/53014693/148972491-018841c1-206a-472a-b14d-a3451c312b14.png)
Focus is on the last element of the page. It should be on the next element related to the dropdown.

State after pressing Shift + TAB twice when the dropdown is opened:
![image](https://user-images.githubusercontent.com/53014693/148972590-f7a7de72-70bc-4e9d-9215-2221bcf3afd1.png)
Focus is on the last element of the page. It should be on the previous element related to the dropdown.

## Expected Behavior

The focus should go to the next/previous element element.

## Environment

* Component Name and Version: terra-form-select v6.35.0
* Browser Name and Version: Firefox 94.0.2 (64-bit)
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile): Windows 10
